packages/eas-cli/src/utils/queries.ts

```diff
@@ -0,0 +1,151 @@
+import { confirmAsync, selectAsync } from '../prompts';
+import uniqBy from './expodash/uniqBy';
+
+const fetchMoreValue = '_fetchMore';
+
+export type PaginatedQueryResponse<QueryReturnType extends Record<string, any>> = {
+  queryResponse: QueryReturnType[];
+  queryResponseRawLength: number;
+};
+
+type BasePaginatedQueryArgs<QueryReturnType extends Record<string, any>> = {
+  limit: number;
+  offset: number;
+  queryToPerform: (
+    limit: number,
+    offset: number
+  ) => Promise<PaginatedQueryResponse<QueryReturnType>>;
+};
+
+type PaginatedQueryWithConfirmPromptArgs<QueryReturnType extends Record<string, any>> =
+  BasePaginatedQueryArgs<QueryReturnType> & {
+    promptOptions: {
+      readonly title: string;
+      renderListItems: (currentPage: QueryReturnType[]) => void;
+    };
+  };
+
+type PaginatedQueryWithSelectPromptArgs<QueryReturnType extends Record<string, any>> =
+  BasePaginatedQueryArgs<QueryReturnType> & {
+    promptOptions: {
+      readonly title: string;
+      createDisplayTextForSelectionPromptListItem: (queryItem: QueryReturnType) => string;
+      getIdentifierForQueryItem: (queryItem: QueryReturnType) => string;
+    };
+  };
+
+export async function paginatedQueryWithConfirmPromptAsync<
+  QueryReturnType extends Record<string, any>
+>({
+  limit,
+  offset,
+  queryToPerform,
+  promptOptions,
+}: PaginatedQueryWithConfirmPromptArgs<QueryReturnType>): Promise<void> {
+  return await _paginatedQueryWithConfirmPromptInternalAsync(
+    { limit, offset, queryToPerform, promptOptions },
+    []
+  );
+}
+
+// eslint-disable-next-line no-underscore-dangle
+async function _paginatedQueryWithConfirmPromptInternalAsync<
+  QueryReturnType extends Record<string, any>
+>(
+  {
+    limit,
+    offset,
+    queryToPerform,
+    promptOptions,
+  }: PaginatedQueryWithConfirmPromptArgs<QueryReturnType>,
+  accumulator: QueryReturnType[]
+): Promise<void> {
```

Comment:
in order to not expose the accumulator to API consumers, I removed it from the exposed function's args and made an internal function that allows for an accumulator to be provided.

I used the prefix `_` and sorta-suffix `Internal` to help make it clear what these functions are for. Let me know if you want me to remove the `_`, i thought it made the code clearer

Reply:
Yes, I think that I didn't suggest changing the default argument.